libs/gi/ui/src/components/artifact/editor/index.tsx

```diff
@@ -124,6 +142,38 @@ function artifactReducer(
           ]
         return { ...state! }
       }
+      case 'unactivatedSubstat': {
```

Comment:
changing the 3rd substats key on a 4-line artifact is changing the 4th substats key for some reason